app/server/lib/Authorizer.ts

```diff
@@ -203,6 +203,15 @@ export async function addRequestUser(
       if (!user) {
         return res.status(401).send('Bad request: invalid API key');
       }
+      if (user.type === "service") {
+        const serviceAccount = (await dbManager.getServiceAccountWithOwner(user.loginEmail as string))!;
+        if (serviceAccount.owner.disabledAt) {
+          return res.status(403).send('Unauthorized');
+        }
+        if (!serviceAccount.isAlive()) {
+          return res.status(401).send('Service Account has reached its end of life');
```

Comment:
This case seems to call for a `403` rather than a `401` to me, since the authentication is valid (the API key was found in the database) but the account is unavailable.